src/net/posix-stack.cc

```diff
@@ -871,7 +871,14 @@ class posix_datagram : public udp_datagram_impl {
     posix_datagram(const socket_address& src, const socket_address& dst, packet p) : _src(src), _dst(dst), _p(std::move(p)) {}
     virtual socket_address get_src() override { return _src; }
     virtual socket_address get_dst() override { return _dst; }
-    virtual uint16_t get_dst_port() override { return _dst.port(); }
+    virtual uint16_t get_dst_port() override {
+        if (_dst.family() != AF_INET && _dst.family() != AF_INET6) {
+            std::stringstream ss;
+            ss << _dst;
+            throw std::runtime_error(format("get_dst_port() called on non-IP address: %s", ss.str()));
```

Comment:
I don't think you need this stringstream stuff, format should be able to do it itself. Something like:
```c++
throw std::runtime_error(format("get_dst_port() called on non-IP address: {}",_dst));
```